src/python/bcc/__init__.py

```diff
@@ -1093,7 +1093,12 @@ def cleanup(self):
             # non-string keys here include the perf_events reader
             if isinstance(k, str):
```

Comment:
while looking into it recently I noticed that we waste 8 kernel pages for every kprobe due to this mmap and there are a bunch of kprobes, so make it optional would be good memory savings.